### PR TITLE
feat: publish with the pypa action

### PR DESCRIPTION
**Pull request by razvan**
Needed for attestation support.

Followed instructions from [here](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/) minus package signing.

### CODE REVIEW

File: .github/workflows/pr_pre-commit.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.11", "3.12"]
+        python-version: ["3.11", "3.12", "3.13"]
```
Comment (Techassi) [inline .github/workflows/pr_pre-commit.yml:12]: suggestion: We could use the [`stackabletech/actions/run-pre-commit`](https://github.com/stackabletech/actions/tree/main/run-pre-commit) action here. But it doesn't offer a huge benefit compared to the current solution.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -8,25 +8,41 @@ on:
...
-          twine check dist/*
-      - name: Publish
-        run: |
-          twine upload  --non-interactive --verbose --username __token__ --password ${{ secrets.PYPI_API_TOKEN }} dist/*
+      - name: Store wheel
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882  # v4.4.3
+        with:
+          name: beku-stackabletech-package
+          path: dist/
+  pypi-publish:
```
Comment (Techassi) [inline .github/workflows/publish.yml:31]: question: I'm curious why we split this up into two different jobs? Couldn't we publish the package to PyPi in the same job which would save us the artifact up/download?

**Review by Techassi — COMMENTED** [review]
LGTM, just two comments.

Also, while at it, we should bump the action versions. I think there are a bunch with more recent versions.

File: .github/workflows/pr_pre-commit.yml (hunk 1 above)
Comment (razvan) [reply .github/workflows/pr_pre-commit.yml:12]: [done](https://github.com/stackabletech/beku.py/pull/33/commits/c3650ef58c531582b6018821c0e55e56daa0d6e8)

File: .github/workflows/publish.yml (hunk 2 above)
Comment (Techassi) [reply .github/workflows/publish.yml:31]: Alright, makes sense. Thank you!

**Review by Techassi — APPROVED** [review]

**Pull request merged**